node/src/components/consensus/highway_core/state.rs

```diff
@@ -868,24 +873,55 @@ impl<C: Context> State<C> {
     /// Returns the panorama of the confirmation for the leader unit `vhash`.
     pub(crate) fn confirmation_panorama(
         &self,
-        own_idx: ValidatorIndex,
+        creator: ValidatorIndex,
         vhash: &C::Hash,
     ) -> Panorama<C> {
-        let unit = self.unit(vhash);
-        let mut panorama;
-        // TODO(HWY-167): Confirmation panorama.
-        if let Some(prev_hash) = self.panorama().get(own_idx).correct().cloned() {
-            let own_unit = self.unit(&prev_hash);
-            panorama = unit.panorama.merge(self, &own_unit.panorama);
-            panorama[own_idx] = Observation::Correct(prev_hash);
-        } else {
-            panorama = unit.panorama.clone();
+        self.to_valid_panorama(creator, self.inclusive_panorama(vhash))
+    }
+
+    /// Creates a panorama that is valid for use in `creator`'s next unit, and as close as possible
+    /// to the given one. It is only modified if necessary for validity:
+    /// * Cite `creator`'s previous unit, i.e. don't equivocate.
+    /// * Satisfy the LNC, i.e. don't add new naively cited forks.
+    pub(crate) fn to_valid_panorama(
```

Comment:
What's `to_` prefix for? How should I interpret it?

Reply:
Good point… I was meaning it converts `pan` to a _valid_ panorama, but since it doesn't apply to `self` I'll probably just remove the `to_`. :+1: